src/protocols/fp-wal.ts

```diff
@@ -0,0 +1,145 @@
+import { FPEnvelopeTurnaround, FPEnvelopeType, FPStoreType } from "./fp-envelope.js";
+import { CID } from "multiformats";
+import { WALState } from "../blockstore/types.js";
+
+export type FPGetWALReq = WALState
+/*
+ * A request to get a WAL file.
+ * As JSON, this is:
+ * {
+ *    "type": "get-wal-req",
+ *    "tid": "123",
+ *    "payload": {
+ *   operations: DbMeta[];
```

Comment:
right now this means we need a WAL per remote. we could keep high water marks, but that would require a function from (encrypted) CRDT log events to carLog CIDs. Not hard to do ...  but wait til after we have the next layer of testing.

Reply:
this might be a good place for @damienkatz help